### PR TITLE
[New Rule] Okta User Attempted Unauthorized Access

### LINKED ISSUE #1207: [New Rule] Okta User Attempted Unauthorized Access
## Description

Identifies when a user account attempted to access an Okta application in an unauthorized way.

### Target indexes

```
`filebeat-*, logs-okta*`
```

### Platforms

`Okta`

### Query

```
event.module:okta.system and event.action:app.generic.unauth_app_access_attempt or 
event.message:"User attempted unauthorized access to app"
```

### **False Positives**

- 
- 

### MITRE

| Tactic               | Technique ID | Technique Name | Sub-Technique Name |
| -------------------- | ------------ | -------------- | ------------------ |
| Initial Access       | T1078        | Valid Accounts |                    |
| Defense Evasion      |              |                |                    |
| Persistence          |              |                |                    |
| Privilege Escalation |              |                |                    |

### References
- https://www.okta.com/incident-response-guide/

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1207

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@
 /*.yml
 /config
 /junit.xml
+/elastic-detection-cli
```
Comment (brokensound77) [inline .gitignore:11]:
```suggestion
```

File: rules/okta/privilege_escalation_okta_user_attempted_unauthorized_access.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2021/05/14"
+maturity = "development"
```
Comment (brokensound77) [inline rules/okta/privilege_escalation_okta_user_attempted_unauthorized_access.toml:3]:
```suggestion
maturity = "production"
```

File: rules/okta/privilege_escalation_okta_user_attempted_unauthorized_access.toml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+updated_date = "2021/05/14"
+
+[rule]
+author = ["Elastic", "Austin Songer"]
+description = "Identifies when a user account attempted to access an Okta application in an unauthorized way."
+index = ["filebeat-*", "logs-okta*"]
+language = "kuery"
+name = "Okta User Attempted Unauthorized Access"
+risk_score = 21
```
Comment (brokensound77) [inline rules/okta/privilege_escalation_okta_user_attempted_unauthorized_access.toml:12]:
```suggestion
note = """## Config

The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
risk_score = 21
```

**Comment by threat-punter** [comment]
I see the tests are failing with the error below. @austinsonger do we need the `event.message` field or does `event.action:app.generic.unauth_app_access_attempt` match on the intended behavior on its own?

image
Source: https://developer.okta.com/docs/reference/api/event-types/#catalog

Error:
```
Unknown field
event.dataset:okta.system and event.action:app.generic.unauth_app_access_attempt or event.message:"User attempted unauthorized access to app"
                                                                                    ^^^^^^^^^^^^^

Try adding event.module or event.dataset to specify beats module

stack: 7.13.0, beats: 7.13.2, ecs: 1.9.0
Error: Process completed with exit code 1.
```

**Review by threat-punter — APPROVED** [review]

**Comment by threat-punter** [comment]
I think it looks good @austinsonger! Thanks 😄 

@brokensound77 I made some tweaks to the rule to enable it to pass. Would you mind giving it the once over before it's merged please? 🍻

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**